pkg/workflows/workflows_test.go

```diff
@@ -1,18 +1,191 @@
 package workflows
 
 import (
+	"errors"
+	"fmt"
 	"io"
+	"io/fs"
 	"io/ioutil"
 	"os"
 	"testing"
+	"testing/fstest"
 
 	"github.com/stretchr/testify/assert"
 	appsv1 "k8s.io/api/apps/v1"
 	"k8s.io/client-go/kubernetes/scheme"
 
+	"github.com/Azure/draft/pkg/config"
+	"github.com/Azure/draft/pkg/embedutils"
 	"github.com/Azure/draft/pkg/templatewriter/writers"
+	"github.com/Azure/draft/template"
 )
 
+func TestCreateWorkflows(t *testing.T) {
+	dest := "."
+	deployType := "helm"
+	flagVariables := []string{}
+	templatewriter := &writers.LocalFSWriter{}
+	flagValuesMap := map[string]string{"AZURECONTAINERREGISTRY": "testAcr", "CONTAINERNAME": "testContainer", "RESOURCEGROUP": "testRG", "CLUSTERNAME": "testCluster", "BRANCHNAME": "testBranch"}
+	err := createTempDeploymentFile("charts", "charts/production.yaml", "../../test/templates/helm/charts/production.yaml")
+	assert.Nil(t, err)
+	assert.Nil(t, CreateWorkflows(dest, deployType, flagVariables, templatewriter, flagValuesMap))
+	os.RemoveAll("charts")
+	os.RemoveAll(".github")
+
+	deployType = "kustomize"
+	err = createTempDeploymentFile("overlays/production", "overlays/production/deployment.yaml", "../../test/templates/kustomize/overlays/production/deployment.yaml")
+	assert.Nil(t, err)
+	assert.Nil(t, CreateWorkflows(dest, deployType, flagVariables, templatewriter, flagValuesMap))
+	os.RemoveAll("overlays")
+	os.RemoveAll(".github")
+
+	deployType = "manifests"
+	err = createTempDeploymentFile("manifests", "manifests/deployment.yaml", "../../test/templates/manifests/manifests/deployment.yaml")
+	assert.Nil(t, err)
+	assert.Nil(t, CreateWorkflows(dest, deployType, flagVariables, templatewriter, flagValuesMap))
+	os.RemoveAll("manifests")
+	os.RemoveAll(".github")
+	//test for missing deployment file path
+	assert.NotNil(t, CreateWorkflows(dest, deployType, flagVariables, templatewriter, flagValuesMap))
+
+	//test for invalid deployType
+	deployType = "testInvalidDeployType"
+	assert.NotNil(t, CreateWorkflows(dest, deployType, flagVariables, templatewriter, flagValuesMap))
+  
+}
+
+func TestUpdateProductionDeployments(t *testing.T) {
+	flagValuesMap := map[string]string{"AZURECONTAINERREGISTRY": "testRegistry", "CONTAINERNAME": "testContainer"}
+	testTemplateWriter := &writers.LocalFSWriter{}
+	//test for missing deploy type
+	assert.Nil(t, updateProductionDeployments("", ".", flagValuesMap, testTemplateWriter))
+
+	//test for missing helm deployment file
+	assert.NotNil(t, setHelmContainerImage("", "testImage", testTemplateWriter))
```

Comment:
do all these sub-tests have some shared state? otherwise we could just break this up into several smaller tests instead of one big one